### PR TITLE
Mandatory yearly gun pr: 2020 edition

### PR DESCRIPTION
**Pull request by Alberyk**
-adds the harpoon gun, to the merchant list and as a raider weapon
-adds the foldable shotgun, to merchant list and as a raider weapon
-adds a new sprite and firing sound for the silenced pistol, giving it to the dpra consular as well
-makes harpoons better as a throwing weapon
-tweaks some guns' names

### CODE REVIEW

File: code/modules/projectiles/guns/launcher/harpoon.dm [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+/obj/item/gun/launcher/harpoon/attackby(obj/item/I as obj, mob/user as mob)
+	if(istype(I, /obj/item/material/harpoon))
+		if(harpoons.len < max_harpoons)
+			user.drop_from_inventory(I,src)
+			harpoons += I
+			to_chat(user, "<span class='notice'>You load \the [I] in \the [src].</span>")
+			update_icon()
+		else
+			to_chat(usr, "<span class='warning'>\The [src] is already loaded.</span>")
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/launcher/harpoon.dm]:
```suggestion
			to_chat(user, "<span class='warning'>\The [src] is already loaded.</span>")
```

File: code/modules/projectiles/guns/launcher/harpoon.dm [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	fire_sound = 'sound/weapons/crossbow.ogg'
+	fire_sound_text = "a metallic thunk"
+	recoil = 1
+
+	description_fluff = "Adapted from fishing tools, Frozen Sea Harpoon Guns were found to be useful weapons in the hands of the Royal Navy and Army. Their ammunition can be equipped \
+	with all shorts of implements, such as explosives and even grappling hooks. Large guns are usually mounted on warships, serving as a complement to their traditional cannons. \
+	While smaller version are carried by sailors and fishermen."
+
+	var/max_harpoons = 1
+	var/list/harpoons = new/list()
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/launcher/harpoon.dm]:
```suggestion
	var/list/harpoons = list()
```

File: code/modules/projectiles/guns/launcher/harpoon.dm [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		item_state = "[initial(icon_state)]-[harpoons.len]"
+	update_held_icon()
+
+/obj/item/gun/launcher/harpoon/special_check(mob/user)
+	if(!wielded)
+		to_chat(user, "<span class='warning'>You can't fire without stabilizing \the [src]!</span>")
+		return FALSE
+	return ..()
+
+/obj/item/gun/launcher/harpoon/attackby(obj/item/I as obj, mob/user as mob)
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/guns/launcher/harpoon.dm]:
```suggestion
/obj/item/gun/launcher/harpoon/attackby(obj/item/I, mob/user)
```

File: code/modules/projectiles/guns/projectile/rifle.dm [hunk 4]
```diff
@@ -20,7 +20,7 @@
 	sawnoff_workmsg = "shorten the barrel and stock"
 
 /obj/item/gun/projectile/shotgun/pump/rifle/saw_off(mob/user, obj/item/tool)
-	icon_state = "obrez"
+	icon_state = "sawn-off bolt action rifle"
```
Comment (Arrow768) [inline code/modules/projectiles/guns/projectile/rifle.dm]: icon state

**Pull request merged**